utils/array_utils.h

```diff
@@ -199,11 +210,20 @@ void randint(std::array<numtype, size> &a, std::mt19937 &gen, int min = -1,
  * @param mean The mean of the Gaussian distribution.
  * @param std The standard deviation of the Gaussian distribution.
  */
-void randn(float *a, size_t N, std::mt19937 &gen, float mean = 0.0,
-           float std = 1.0) {
-  std::normal_distribution<float> dist(mean, std);
+template<class precision=float>
```

Comment:
i think instead of committing to implement variants of every function here for each precision type `randn`, `transpose`, etc. let's start with a clean f32 implementation with the expectation that most tasks can be handled by deferring the casting to after any AOT host operations are.

`randn`, `transpose` are already written with the expectation that they're only used for testing or non-critical AOT data preparation. afaict implementing support for half here doesn't add much either in performance or what's possible over deferring the half cast until other CPU preparation/testing code is done in f32.

If I'm missing something let me know.